src/studio/src/designer/backend/Infrastructure/GitRepository/AltinnAppGitRepository.cs

```diff
@@ -249,16 +267,38 @@ private static void GetTextResourceForLanguage(Dictionary<string, Dictionary<str
         /// identified by the languageCode in filename.
         /// </summary>
         /// <param name="languageCode">Language identifier</param>
-        /// <returns>Text as a string</returns>
-        public async Task<string> GetTextV2(string languageCode)
+        /// <returns>Texts as a string</returns>
+        public async Task<string> GetTextsV2(string languageCode)
         {
-            string fileName = $"text.{languageCode}.json";
+            string fileName = $"texts.{languageCode}.json";
 
-            var textFileRelativeFilePath = Path.Combine(CONFIG_FOLDER_PATH, LANGUAGE_RESOURCE_FOLDER_NAME, fileName);
+            var textsFileRelativeFilePath = Path.Combine(CONFIG_FOLDER_PATH, LANGUAGE_RESOURCE_FOLDER_NAME, fileName);
 
-            string text = await ReadTextByRelativePathAsync(textFileRelativeFilePath);
+            string texts = await ReadTextByRelativePathAsync(textsFileRelativeFilePath);
+
+            return texts;
+        }
+
+        /// <summary>
+        /// Deletes the texts file for a specific languageCode.
+        /// </summary>
+        /// <param name="languageCode">Language identifier</param>
+        /// <returns>A boolean value that indicates if the file was deleted or not</returns>
+        public bool DeleteTexts(string languageCode)
+        {
+            string fileName = $"texts.{languageCode}.json";
+
+            var textsFileRelativeFilePath = Path.Combine(CONFIG_FOLDER_PATH, LANGUAGE_RESOURCE_FOLDER_NAME, fileName);
+
+            bool deleted = false;
+
+            if (File.Exists(GetAbsoluteFilePathSanitized(textsFileRelativeFilePath)))
```

Comment:
Avoid direct file access in this class and use the methods in the GitRepository class instead. There is a FileExistsByRelativePath method that do the same thing. 

No need to check if a file exists before deleting, just delete it and do a try/catch in the controller and return a 404 if it throws a IOException like the Get method does. This shouldn't happen and is an exceptional case. This builds upon methods being either commands or queries. Commands don't return data and have side effects (file deleted). Queries don't have side effects and return data.